### PR TITLE
add html5 form

### PR DESCRIPTION
**Pull request by iamsokol**
html5 form

[Demo](https://iamsokol.github.io/form/) |
[Code base](https://github.com/iamsokol/iamsokol.github.io/tree/master/form)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by iamsokol** [comment]
@A-Ostrovnyy, this is a prototype and not a final decision.

File: submissions/iamsokol/form/index.html [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    <div class="footer-in">
+      <div class="footer-item description">
+        <h3 class="footer-item-title">What we do</h3>
+        <p class="footer-item-description">We provide a full range of language services including content development, document, website and social media accounts translations.</p>
+      </div>
+      <div class="footer-item contact">
+        <h3 class="footer-item-title">Our main office</h3>
+        <p class="footer-item-description">Dubai, UAE</p>
+        <p class="footer-item-description">P.O. Box: 502735</p>
+        <p class="footer-item-description">Phone: +971(0)44278128</p>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Use link `<a href="tel:+971(0)44278128"></a>` for phone numbers.

File: submissions/iamsokol/form/index.html [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+      </div>
+      <div class="footer-item social">
+        <h3 class="footer-item-title">Connect with us</h3>
+        <a href="#" class="footer-item-description"><i class="fa fa-facebook" aria-hidden="true"></i> Facebook</p>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Fix typo: `Facebook </a>`.

File: submissions/iamsokol/form/index.html [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    </button>
+    <img class="header-logo" src="assets/img/logo.png" alt="logo">
+    <button class="header-sign sm-button" type="button" name="button">
+      <i class="fa fa-user" aria-hidden="true"></i>
+      Sign in
+    </button>
+  </header>
+  <main class="main">
+    <form action="#" method="POST" class="form">
+      <section class="main-head">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Recommendation:  in this case, you can use a `header` tag.

File: submissions/iamsokol/form/index.html [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+          <h3 class="order-title">Help center</h3>
+          <p class="help-description">If you need assistance while placing your order, contact one of our beloved Project Manager by phone or email.</p>
+          <div class="help-in">
+            <a href="tel:+9710555555" class="help-number">+971 (0) 555 555</a>
+            <a href="mailto:help@qtest.com" class="help-mail">help@qtest.com</a>
+          </div>
+        </div>
+      </section>
+      <section class="select">
+        <h1 class="select-title">Select what you want to translate</h1>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: `h1` must be one at the page and it better use for the logo.

File: submissions/iamsokol/form/main.scss [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+  padding: 10%;
+  text-align: center;
+}
+.select-type{
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  font-weight: bold;
+  input{
+    display: none;
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Use `.visually-hidden` class from my demo for hiding default controls. In your case click from the label not going to the checkbox.

File: submissions/iamsokol/form/main.scss [hunk 6]
```diff
@@ -0,0 +1,661 @@
...
+  }
+  img{
+    width: 30%;
+    display: block;
+    margin: 0 auto 20px;
+  }
+  label{
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Add styles on `:hover`, and `:focus` state.

File: submissions/iamsokol/form/main.scss [hunk 7]
```diff
@@ -0,0 +1,661 @@
...
+    @include border(#d7d9e0, #6e768f);
+  }
+}
+.title,
+.title + .subtitle{
+  display: inline-block;
+}
+.document{
+  .file{
+    display: none;
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Use `visually-hidden` class for hiding default inputs controls.

File: submissions/iamsokol/form/index.html [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+            <p>Remove</p>
+          </div>
+          <div class="document-in-item">
+            <p>
+              <i class="fa fa-file-word-o" aria-hidden="true"></i>
+              bussiness_operations.docx
+            </p>
+            <p>78kb</p>
+            <p>842</p>
+            <i class="fa fa-trash" aria-hidden="true"></i>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add styles on `:hover`, `:focus`, and `:active` state.

File: submissions/iamsokol/form/index.html [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+          </div>
+          <div class="select-wrapper languageTo2">
+            <select name="languageTo2" id="languageTo2" class="selectElement sm-button">
+              <option value="France">France</option>
+              <option value="Arabic">Arabic</option>
+              <option value="Russian">Russian</option>
+            </select>
+          </div>
+          <i class="fa fa-trash" aria-hidden="true"></i>
+          <button class="sm-button" type="button" name="button">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: In the mockup this button more narrow.

File: submissions/iamsokol/form/index.html [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+          </div>
+        </fieldset>
+        <div class="range-info">
+          <h3 class="range-info-title">Timeline info</h3>
+          <p class="range-info-description">With an <a href="#">avarage</a> timeline set up, our expert translator can take a reasonable amount of time perfecting y our translation</p>
+        </div>
+      </section>
+      <section class="payment">
+        <h2 class="title">Sign up and choose a payment method</h2>
+        <p class="payment-description">Already have account with us? <a href="#">Sign in</a></p>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add styles on `:hover` state to the link.

File: submissions/iamsokol/form/index.html [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+        <fieldset class="full-width">
+          <legend>Full name</legend>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Better use `label` for this text.

File: submissions/iamsokol/form/index.html [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+            </label>
+            <input type="radio" id="amazon" name="payment">
+            <label for="amazon">
+              <img src="assets/img/amazon.png" alt="amazon">
+            </label>
+          </div>
+        </fieldset>
+        <fieldset class="payment-card">
+          <legend>Card number</legend>
+          <input type="number" placeholder="Card number" value="" class="sm-input">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add `background-image`. See mockup.

File: submissions/iamsokol/form/index.html [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+        </fieldset>
+        <fieldset class="payment-expiry">
+          <legend>Expiry date</legend>
+          <input type="date" placeholder="DD / MM / YY" value="" class="sm-input">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Fix placeholder text as in the mockup.

File: submissions/iamsokol/form/index.html [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+        </fieldset>
+        <fieldset class="payment-code">
+          <legend>Code</legend>
+          <input type="password" placeholder="Code" value="" class="sm-input">
+        </fieldset>
+        <div class="payment-submit">
+          <input type="checkbox" name="terms" id="terms">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Fix styles. [Image](https://imgur.com/a/YuHGxpe)

File: submissions/iamsokol/form/index.html (hunk 4 above)
Comment (iamsokol) [reply submissions/iamsokol/form/index.html]: I did not understand this remark)
h1 tag is one on page. wrap the logo in it?

File: submissions/iamsokol/form/index.html (hunk 9 above)
Comment (iamsokol) [reply submissions/iamsokol/form/index.html]: index
index2

Comment (A-Ostrovnyy) [reply submissions/iamsokol/form/index.html]: I mean `+Add` button, sorry for the misunderstanding.

File: submissions/iamsokol/form/index.html (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/iamsokol/form/index.html]: This is one technique: if you have a logo(every site have a logo)) you hide text(usually its company name + small description) inside `h1` tag and use it for the logo.
Wrap logo into `h1` tag.

**Comment by iamsokol** [comment]
@A-Ostrovnyy, I updated my code

File: submissions/iamsokol/form/index.html [hunk 15]
```diff
@@ -14,15 +14,15 @@
       <i class="fa fa-long-arrow-left" aria-hidden="true"></i>
       Go back
     </button>
-    <img class="header-logo" src="assets/img/logo.png" alt="logo">
+    <h1><img class="header-logo" src="assets/img/logo.png" alt="logo"></h1>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: The logo must be a link inside heading.

File: submissions/iamsokol/form/index.html [hunk 16]
```diff
@@ -89,7 +89,7 @@ <h2 class="title">Upload your documents</h2>
             <i class="fa fa-cloud-upload" aria-hidden="true"></i>
             Upload a new file
           </label>
-          <input class="file" type="file" id="file" name="file" title="download file">
+          <input class="file visually-hidden" type="file" id="file" name="file" title="download file">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add this button to the accessibility context.

File: submissions/iamsokol/form/index.html [hunk 17]
```diff
@@ -105,7 +105,7 @@ <h2 class="title">Upload your documents</h2>
             </p>
             <p>78kb</p>
             <p>842</p>
-            <i class="fa fa-trash" aria-hidden="true"></i>
+            <button class="trash" name="button" type="button"><i class="fa fa-trash" aria-hidden="true"></i></button>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add `cursor="pointer"` for better UX.

File: submissions/iamsokol/form/index.html [hunk 18]
```diff
@@ -189,49 +189,56 @@ <h3 class="range-info-title">Timeline info</h3>
       <section class="payment">
         <h2 class="title">Sign up and choose a payment method</h2>
         <p class="payment-description">Already have account with us? <a href="#">Sign in</a></p>
-        <fieldset class="full-width">
-          <legend>Full name</legend>
-          <input type="text" pattern="[A-Za-zА-Яа-яЁё]{1,}" placeholder="Full name" value="Mathieu Mayer" class="sm-input validate">
+        <fieldset class="full-width validate">
+          <label>Full name</label>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: A label must be linked to the input by id.

**Comment by iamsokol** [comment]
@A-Ostrovnyy, I already fixed it

**Comment by A-Ostrovnyy** [comment]
Check validation: [Image](https://imgur.com/a/Vi6DONn)

File: submissions/iamsokol/form/index.html (hunk 4 above)
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Use heading `h2` instead of `p` for defining title

File: submissions/iamsokol/form/main.scss [hunk 19]
```diff
@@ -0,0 +1,661 @@
...
+  padding: 0;
+  box-sizing: border-box;
+  line-height: 1;
+  font-family: arial, sans-serif;
+  font-size: $fontSmall;
+}
+body{
+  background: #eceef4;
+}
+h1{
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Avoid styling by html tag.

File: submissions/iamsokol/form/main.scss [hunk 20]
```diff
@@ -0,0 +1,661 @@
...
+      right: 15px;
+      transform: translateY(-50%);
+      height: 30px;
+      width: 30px;
+      line-height: 30px;
+      text-align: center;
+      background: #76a7e1;
+      color: #fff;
+      border-radius: 30px; 
+      font: normal normal normal 20px/30px FontAwesome;
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: `font` is a deprecated CSS property. Use expanded option properties.

File: submissions/iamsokol/form/main.scss [hunk 21]
```diff
@@ -76,10 +78,71 @@ h3, h4, h5, h6{
...
   padding: 0;
+  @media (max-width: $tablet) {
+    width: 100%;
+  }
 }
+
+.visually-hidden{
+  display: none;
+}
+.visually-radio{
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: [Watch video how to hide default inputs](https://youtu.be/E6kLaaQFctU)

File: submissions/iamsokol/form/index.html [hunk 22]
```diff
@@ -0,0 +1,263 @@
...
+      </section>
+      <section class="document">
+        <h2 class="title">Upload your documents</h2>
+        <p class="subtitle">(Doc, Docx, Text, PDF, Markdown)</p>
+        <fieldset>
+          <label for="file" class="sm-button">
+            <i class="fa fa-cloud-upload" aria-hidden="true"></i>
+            Upload a new file
+          </label>
+          <input class="file" type="file" id="file" name="file" title="download file">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: [Watch video how to hide default inputs](https://youtu.be/E6kLaaQFctU)

File: submissions/iamsokol/form/index.html [hunk 23]
```diff
@@ -0,0 +1,263 @@
...
+        <p class="footer-item-description">P.O. Box: 502735</p>
+        <p class="footer-item-description">Phone: +971(0)44278128</p>
+      </div>
+      <div class="footer-item social">
+        <h3 class="footer-item-title">Connect with us</h3>
+        <a href="#" class="footer-item-description"><i class="fa fa-facebook" aria-hidden="true"></i> Facebook</p>
+        <a href="#" class="footer-item-description"><i class="fa fa-twitter" aria-hidden="true"></i> Twiter</a>
+        <a href="#" class="footer-item-description"><i class="fa fa-linkedin" aria-hidden="true"></i> LinkedIn</a>
+      </div>
+      <img src="assets/img/logo2.png" alt="logo2">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Logo must be a link.

File: submissions/iamsokol/form/index.html (hunk 2 above)
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Add styles on `:hover` effect.

File: submissions/iamsokol/form/index.html [hunk 24]
```diff
@@ -228,15 +228,15 @@ <h2 class="title">Sign up and choose a payment method</h2>
         </fieldset>
         <fieldset class="payment-card">
           <legend>Card number</legend>
-          <input type="number" placeholder="Card number" class="sm-input card-number">
+          <input type="text" pattern="[0-9]{16}" required placeholder="Card number" class="sm-input card-number" title="Set card 16 numbers">
         </fieldset>
         <fieldset class="payment-expiry">
           <legend>Expiry date</legend>
-          <input type="date" placeholder="DD / MM / YY" class="sm-input">
+          <input type="date" placeholder="DD / MM / YY" required class="sm-input" title="Enter the correct format">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Write example of data in the `title` attribute.

File: submissions/iamsokol/form/main.scss [hunk 25]
```diff
@@ -0,0 +1,661 @@
...
+  }
+  .document-in{
+    padding: 20px 60px;
+    border: 1px solid #eef0f4;
+    border-radius: 5px;
+    div{
+      display: flex;
+      flex-wrap: wrap;
+      padding: 20px 0;
+      p{
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Avoid creating/using too specific selectors.

File: submissions/iamsokol/form/main.scss [hunk 26]
```diff
@@ -0,0 +1,661 @@
...
+      appearance: none;
+      height: 20px;
+      width: 20px;
+      background-color: #edbe76;
+      box-sizing: border-box;
+      z-index: 2;
+      border-radius: 100%;
+      top: -50%;
+  }
+  &::-webkit-slider-runnable-track {
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/main.scss]: Add `border-radius`. [Image](https://imgur.com/a/foSzHiC)

File: submissions/iamsokol/form/index.html [hunk 27]
```diff
@@ -0,0 +1,263 @@
...
+          <input type="range" step="10" value="50" name="range" min="0" max="100" id="range" class="range">
+          <div class="range-label">
+            <label for="range">I got time</label>
+            <label for="range">Avarage</label>
+            <label for="range">Yesterday</label>
+          </div>
+        </fieldset>
+        <div class="range-info">
+          <h3 class="range-info-title">Timeline info</h3>
+          <p class="range-info-description">With an <a href="#">avarage</a> timeline set up, our expert translator can take a reasonable amount of time perfecting y our translation</p>
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: In the mockup, this text has 2 lines

File: submissions/iamsokol/form/index.html [hunk 28]
```diff
@@ -0,0 +1,263 @@
...
+        </div>
+      </section>
+      <section class="payment">
+        <h2 class="title">Sign up and choose a payment method</h2>
+        <p class="payment-description">Already have account with us? <a href="#">Sign in</a></p>
+        <fieldset class="full-width">
+          <legend>Full name</legend>
+          <input type="text" pattern="[A-Za-zА-Яа-яЁё]{1,}" placeholder="Full name" value="Mathieu Mayer" class="sm-input validate">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: The image must be hidden by default and show to the user when the value of the input value valid or invalid.

File: submissions/iamsokol/form/index.html [hunk 29]
```diff
@@ -0,0 +1,263 @@
...
+      </section>
+    </form>
+  </main>
+  <footer class="footer">
+    <div class="footer-in">
+      <div class="footer-item description">
+        <h3 class="footer-item-title">What we do</h3>
+        <p class="footer-item-description">We provide a full range of language services including content development, document, website and social media accounts translations.</p>
+      </div>
+      <div class="footer-item contact">
```
Comment (A-Ostrovnyy) [inline submissions/iamsokol/form/index.html]: Mabey for this case better to use `address` tag.

File: submissions/iamsokol/form/index.html (hunk 27 above)
Comment (iamsokol) [reply submissions/iamsokol/form/index.html]: I did not understand the comment. This text have 2 lines

Comment (A-Ostrovnyy) [reply submissions/iamsokol/form/index.html]: Maybe it's because of my display to narrow. [Image](https://imgur.com/a/eOEVcHu)

Comment (iamsokol) [reply submissions/iamsokol/form/index.html]: Is an increasing of the lines while screen decreasing - a normal behavior?
image

**Comment by A-Ostrovnyy** [comment]
Good job!
Fix the mobile view. [Image](https://imgur.com/a/1RyzzVS)
Also, add styles on `:hover` to the logos.

File: submissions/iamsokol/form/index.html (hunk 27 above)
Comment (A-Ostrovnyy) [reply submissions/iamsokol/form/index.html]: Yes, it's normal behavior. The text needs to go somewhere.)

**Comment by iamsokol** [comment]
@A-Ostrovnyy I fixed

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**